src/execution_plan/ops/op_update.c

```diff
@@ -213,18 +256,17 @@ OpResult OpUpdateReset(OpBase *ctx) {
 
 void OpUpdateFree(OpBase *ctx) {
     OpUpdate *op = (OpUpdate*)ctx;
-
-    /* Lock everything. */
-    Graph_AcquireWriteLock(op->gc->g);
-    _CommitUpdates(op);
-    // Release lock.
-    Graph_ReleaseLock(op->gc->g);
-    
     /* Free each update context. */
     for(uint i = 0; i < op->update_expressions_count; i++) {
         AR_EXP_Free(op->update_expressions[i].exp);
     }
 
+    if(op->records) {
+        uint records_count = array_len(op->records);
+        for(uint i = 0; i < records_count; i++) Record_Free(op->records[i]);
```

Comment:
I think this for loop will always be a no-op, since we should have always popped every record in the `_handoff` logic. That said, it might be safer to keep it - perhaps just add a comment to document this.

Reply:
Consider `MATCH (n) SET n.v = 1 RETURN n LIMIT 1`
In the case where there are multiple `n` nodes we'll update all of them (an optimisation on the size of records can be applied here) but we'll only pop one record as the LIMIT will be reached, as a result `records` will not be empty.